UserService/UserService.UnitTests/GetHelpersByPostcodeAndTaskTypeHandlerTests.cs

```diff
@@ -0,0 +1,194 @@
+using UserService.Core.Utils;
+using UserService.Handlers;
+
+namespace UserService.UnitTests
+{
+    public class GetHelpersByPostcodeAndTaskTypeHandlerTests
+    {
+        private Mock<IVolunteerCache> _volunteerCache;
+        private Mock<IDistanceCalculator> _distanceCalculator;
+        private Mock<IAddressService> _addressService;
+        private IHelperService _helperService;
+        private Mock<IRepository> _repository;
+
+        private IEnumerable<CachedVolunteerDto> _cachedVolunteerDtos;
+
+        private GetPostcodeCoordinatesResponse _getPostcodeCoordinatesResponse;
+
+        private IEnumerable<User> _users;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _cachedVolunteerDtos = new List<CachedVolunteerDto>()
+            {
+                new CachedVolunteerDto()
+                {
+                    UserId = 1,
+                    Postcode = "NG1 1AA",
+                    VolunteerType = VolunteerType.Helper,
+                    IsVerifiedType = IsVerifiedType.IsVerified,
+                    Latitude = 1,
+                    Longitude = 2,
+                    SupportRadiusMiles = 2
+                },
+                new CachedVolunteerDto()
+                {
+                    UserId = 2,
+                    Postcode = "NG1 1AB",
+                    VolunteerType = VolunteerType.Helper,
+                    IsVerifiedType = IsVerifiedType.IsVerified,
+                    Latitude = 3,
+                    Longitude = 4,
+                    SupportRadiusMiles = 1.9
+                },
+            };
+
+            _repository = new Mock<IRepository>();
+
+            _volunteerCache = new Mock<IVolunteerCache>();
+            _volunteerCache.Setup(x => x.GetCachedVolunteersAsync(It.IsAny<VolunteerType>(), It.IsAny<IsVerifiedType>(), It.IsAny<CancellationToken>())).ReturnsAsync(_cachedVolunteerDtos);
+
+            _distanceCalculator = new Mock<IDistanceCalculator>();
+
+            _distanceCalculator.Setup(x => x.GetDistanceInMiles(It.IsAny<double>(), It.IsAny<double>(), It.IsAny<double>(), It.IsAny<double>())).Returns(2);
+
+
+            _getPostcodeCoordinatesResponse = new GetPostcodeCoordinatesResponse()
+            {
+                PostcodeCoordinates = new List<PostcodeCoordinate>()
+                {
+                    new PostcodeCoordinate()
+                    {
+                        Postcode = "NG1 1AE",
+                        Latitude = 9,
+                        Longitude = 10
+                    }
+                }
+            };
+
+            _addressService = new Mock<IAddressService>();
+
+            _addressService.Setup(x => x.GetPostcodeCoordinatesAsync(It.IsAny<GetPostcodeCoordinatesRequest>(), It.IsAny<CancellationToken>())).ReturnsAsync(_getPostcodeCoordinatesResponse);
+            _helperService = new HelperService(_addressService.Object, _volunteerCache.Object, _distanceCalculator.Object, _repository.Object);
```

Comment:
I would have separate tests for the HelperService rather than have them hidden away in the tests for GetHelpersByPostcodeAndTaskTypeHandler as it's a crucial piece of logic that is used by two handlers and is very likely to be used again by another handler.  It should be easy to refactor out and then mock the call to IHelperService in the tests for GetHelpersByPostcodeAndTaskTypeHandler.

Reply:
your wish is my command